spec/struct.dd

```diff
@@ -62,7 +62,9 @@ $(H2 $(LNAME2 struct_layout, Struct Layout))
     the first field and the start of the object.
     )
 
-    $(P Structs with no fields of non-zero size (aka $(I Empty Structs)) have a size of one byte.)
+    $(P A struct with no fields of non-zero size is an $(I Empty Struct).
+    An Empty Struct with a $(DDSUBLINK spec/attribute, linkage, Linkage Attribute) of `extern (C)` has a size of zero.
+    Otherwise, its size is one byte.)
```

Comment:
This is wrong. I think you wanted to say that empty `extern (C) struct` size is the same as with the underlying C compiler. Also you might want to add what happens if the C compiler simply disallows empty structs?

Reply:
C11 6.7.2.1-8 says "If the struct-declaration-list does not contain any named members, either directly or via an anonymous structure or anonymous union, the behavior is undefined."

Undefined behavior is not disallowed.

Reply:
If we encounter a C compiler with different behavior, we'll adjust the spec.

Reply:
From what I recall, the problem being that gcc and clang don't agree with each other _on the same platform_. So you could choose to imitate gcc on Linux, but then you're abi incompatible with clang on Linux - which someone might instead choose to use.

But that's all by the by.

Reply:
> Undefined behavior is not disallowed.

The point is, undefined behaviour allows refusal to compile. A reasonable C compiler could emit an error when encountering an empty struct.

> If we encounter a C compiler with different behavior, we'll adjust the spec.

Better, but why not do it right away?